kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -764,6 +809,141 @@ namespace Kratos
         return is_shared;
     }
 
+    template<std::size_t TDim>
+    void CalculateDiscontinuousDistanceToSkinProcess<TDim>::CheckAndCorrectEdgeDistances()
+    {
+
+        KRATOS_TRY;
+
+        CalculateElementalNeighbours();
+
+        auto p_global_ptr_comm = CreatePointerCommunicator();
+
+        // Proxy to retrieve cut edges from neighbours in other partitions. Works as a lambda in serial runs.
+        auto get_neighbour_cut_edges = p_global_ptr_comm->Apply([](GlobalPointer<Element> pElem) {
+
+            std::unordered_map<IndexType, IndexType> global_id_to_local;
+            for (IndexType i = 0; i < mNumNodes; ++i) {
+                global_id_to_local[pElem->GetGeometry()[i].Id()] = i;
+            }
+
+            auto elem_dist = pElem->GetValue(ELEMENTAL_DISTANCES);
+            std::vector<std::vector<IndexType>> cut_edges_vector;
+            const auto edges_container = pElem->GetGeometry().GenerateEdges();
+            for (std::size_t i_edge = 0; i_edge < mNumEdges; ++i_edge) {
+                IndexType edge_i_id = edges_container[i_edge][0].Id();
+                IndexType edge_j_id = edges_container[i_edge][1].Id();
+                double i_distance = elem_dist[global_id_to_local[edge_i_id]];
+                double j_distance = elem_dist[global_id_to_local[edge_j_id]];
+                bool is_edge_cut = (i_distance * j_distance) < 0.0;
+                std::vector<IndexType> edge_vector;
+                edge_vector.push_back(std::min(edge_i_id, edge_j_id));
+                edge_vector.push_back(std::max(edge_i_id, edge_j_id));
+                if (is_edge_cut) {
+                    cut_edges_vector.push_back(edge_vector);
+                }
+            }
+            return cut_edges_vector;
+        });
```

Comment:
@roigcarlo here is the usage of the global pointers. We are interested in checking if an edge that was initially identified as intersected is still intersected (after the distance correction added in this PR). This is done by checking if any of the neighbour elements are reporting the edge as intersected by checking their distance values. If none of the elements have the edge as intersected, the initially cut edge is considered as not intersected.